pandas/tools/util.py

```diff
@@ -61,6 +61,13 @@ def to_numeric(arg, errors='raise'):
         - If 'raise', then invalid parsing will raise an exception
         - If 'coerce', then invalid parsing will be set as NaN
         - If 'ignore', then invalid parsing will return the input
+    downcast : str, numpy.dtype, or Python dtype, default None
```

Comment:
@gfyoung , I think this could be used to compact OR to downcast, they are conceptually very similar, compact is saying give me the smallest int, while downcast is saying be a np.int8. I would make 'int', 'unsigned', 'float' do compacting, while a more specific dtype do an astype. 

This makes ``pd.to_numeric`` a fairly versatile routine.